src/arkouda_server.chpl

```diff
@@ -290,118 +294,65 @@ proc main() {
-                when "register"          {repTuple = registerMsg(cmd, args, st);}
-                when "attach"            {repTuple = attachMsg(cmd, args, st);}
-                when "unregister"        {repTuple = unregisterMsg(cmd, args, st);}
-                when "clear"             {repTuple = clearMsg(cmd, args, st);}               
-                when "connect" {
-                    if authenticate {
-                        repTuple = new MsgTuple("connected to arkouda server tcp://*:%i as user %s with token %s".format(
-                                                          ServerPort,user,token), MsgType.NORMAL);
-                    } else {
-                        repTuple = new MsgTuple("connected to arkouda server tcp://*:%i".format(ServerPort), 
-                                                                                        MsgType.NORMAL);
-                    }
-                }
-                when "disconnect" {
-                    repTuple = new MsgTuple("disconnected from arkouda server tcp://*:%i".format(ServerPort), 
-                                                                   MsgType.NORMAL);
-                }
-                when "noop" {
-                    repTuple = new MsgTuple("noop", MsgType.NORMAL);
-                }
-                when "ruok" {
-                    repTuple = new MsgTuple("imok", MsgType.NORMAL);
-                }
-                otherwise {
-                    repTuple = new MsgTuple("Unrecognized command: %s".format(cmd), MsgType.ERROR);
-                    asLogger.error(getModuleName(),getRoutineName(),getLineNumber(),repTuple.msg);
-                }
-            }
+            
+            if commandMap.contains(cmd) then
+              repTuple = commandMap.getBorrowed(cmd)(cmd, args, st);
+            else {
+              select cmd
+              {
+                  when "array"             {repTuple = arrayMsg(cmd, args, payload, st);}
+                  when "tondarray"         {
+                    var binaryRepMsg = tondarrayMsg(cmd, args, st);
+                    sendRepMsg(binaryRepMsg);
+                  }
+                  // TODO: this requires SegmentedMsg to be in included
+                  // in the build always; not sure best way to handle
+                  when "segStr-tondarray"  {
+                    use SegmentedMsg;
+                    var binaryRepMsg = segStrTondarrayMsg(cmd, args, st);
+                    sendRepMsg(binaryRepMsg);
+                  }
+                  when "lshdf"             {repTuple = lshdfMsg(cmd, args, st);}
+                  when "readAllHdf"        {repTuple = readAllHdfMsg(cmd, args, st);}
+                  when "tohdf"             {repTuple = tohdfMsg(cmd, args, st);}
+                  when "readAllParquet"    {repTuple = readAllParquetMsg(cmd, args, st);}
+                  when "writeParquet"      {repTuple = toparquetMsg(cmd, args, st);}
```

Comment:
The parquet support should probably be moved into the ServerModules.cfg and eventually the HDF5 support.

Reply:
The reason that was left out of this initial work is that, today, the Parquet message is in `GenSymIO`, so we would need to move it out to it's own module to make it optional. We could include that in this PR, but I think it would be a good idea to do that separately, either in https://github.com/Bears-R-Us/arkouda/pull/1007, or I could work on another branch that does just that.

What are you thoughts on that? Would you rather include the Parquet/HDF5 separation in this PR or another one?

Reply:
Ah, that makes sense, I'd leave it as is.  Once we get a better idea at how the I/O abstraction is going to work we can roll it as part of that process.